lib/SPIRV/SPIRVReader.cpp

```diff
@@ -2681,6 +2681,26 @@ Function *SPIRVToLLVM::transFunction(SPIRVFunction *BF) {
     return Loc->second;
 
   auto IsKernel = isKernel(BF);
+
+  if (IsKernel) {
```

Comment:
Maybe we don't need this step at all? I guess inline optimizations should be able to deal with it.

Reply:
At least for the SPIRV-LLVM-Translator tests we need it, since they don't get opt passes.